apps/frontend/src/utils/question/index.ts

```diff
@@ -269,21 +248,127 @@ export const recallOptions = (
+            name: "city",
+            isNumerical: false,
+            type: "PERSON",
+            propertyType: "String",
+            __typename: "PropertyDefinition",
+        },
+        __typename: "PropertyDefinitionCountableEdge",
+    },
+    {
+        node: {
+            id: "UHJvcGVydHlEZWZpbml0aW9uOjAxOGM3MzVlLTkwZmMtMDAwMy04ZDc5LWEwNGI4NjdlYzM4YQ==",
+            name: "age",
+            isNumerical: true,
+            type: "PERSON",
+            propertyType: "Numeric",
+            __typename: "PropertyDefinition",
+        },
+        __typename: "PropertyDefinitionCountableEdge",
     },
     {
-        label: "last name",
-        value: "last name",
+        node: {
+            id: "UHJvcGVydHlEZWZpbml0aW9uOjAxOGM3MzVlLTkwZmMtMDAwNC0wMzM3LTE3Y2MzM2MyY2Y3Nw==",
+            name: "enabled",
+            isNumerical: false,
+            type: "PERSON",
+            propertyType: "Boolean",
+            __typename: "PropertyDefinition",
+        },
+        __typename: "PropertyDefinitionCountableEdge",
     },
 ];
+
+const userAttributeOptions = attributes.map((attr) => ({
+    value: attr.node.name,
+    id: "attribute" + " " + `attribute.${attr.node.name}`,
+    type: "userAttribute",
+    // disabled: false,
+}));
+
+export const tagOptions = (questions: ParsedQuestion[], openQuestion: ParsedQuestion): TagOption[] => {
+    const recallOpts = recallOptions(questions, openQuestion);
+    const userAttrOpts = userAttributeOptions;
+
+    if (recallOpts.length === 0) {
+        return [{ id: "UserAttribute", value: "User Attribute", disabled: true, type: "" }, ...userAttrOpts];
+    } else if (userAttrOpts.length === 0) {
+        return [{ id: "recalledQuestion", value: "recall from", disabled: true, type: "" }, ...recallOpts];
+    } else {
+        return [
+            { id: "recalledQuestion", value: "recall from", disabled: true, type: "" },
+            ...recallOpts,
+            { id: "UserAttribute", value: "User Attribute", disabled: true, type: "" },
+            ...userAttrOpts,
+        ];
+    }
+};
+
+function resolveQuestionIndex(questionId: string, tagOptions: TagOption[]): string {
+    const option = tagOptions?.find((o) => {
+        const optionId = o.id.split(" ")[0];
+        return questionId === optionId;
+    });
+
+    return option?.value as string;
+}
+
+export function sendToDB(textContent: string): string {
+    const encodedText = textContent.replace(
+        /<span class="mention"([^>]*)data-id="([^"]*)"([^>]*)data-type="([^"]*)"([^>]*)data-fallback="([^"]*)"([^>]*)>(.*?)<\/span>/g,
+        (_, __, dataId, ___, dataType, ____, fallback) => {
+            if (dataId === "attribute") {
+                return `{{${dataType} | "${fallback}"}}`;
+            }
+            return `{{${dataType}:${dataId} | "${fallback}"}}`;
+        },
+    );
+
+    return encodedText.split("</span>").join("");
+}
+
+export function getfromDB(encodedText: string, tagOptions: TagOption[]): string {
```

Comment:
@afolasope Likewise, this should be `decodeText`. Atoms should be generic as possible and they shouldn't care about the internal details.